Only print stacktrace of conflicting buildsteps if activated

Postremus opened the pull request:
The quarkus-core-builder detects conflicts between buildsteps, and throws exception for that.
This exception also included the stacktrace of the original caller of the build method. However, this pointed to a big lambda chain.
Since this stacktrace was created for each buildstep, this got costly fast.

Now, simply only include by default an empty ST, and add an option to include the full stacktrace.

This saves about 5-10ms of dev mode startup time, on the reproducer from #21552.

Relates to #21552, but again not a complete fix.

gastaldi commented on core/builder/src/main/java/io/quarkus/builder/BuildStepBuilder.java:
```diff
@@ -15,6 +15,11 @@
  * a destructor for items it produces, which will be run (in indeterminate order) at the end of processing.
  */
 public final class BuildStepBuilder {
+    private static final boolean LOG_CONFLICT_CAUSING = Boolean
+            .parseBoolean(System.getProperty("quarkus.builder.log-conflict-cause", "false"));
```
```suggestion
    private static final boolean LOG_CONFLICT_CAUSING = Boolean
            .getBoolean("quarkus.builder.log-conflict-cause");
```

Postremus replied:
Did not know this one existed, thank you!

gastaldi reviewed (approved):
LGTM

gastaldi commented on core/builder/src/main/java/io/quarkus/builder/BuildChainBuilder.java:
```diff
@@ -186,10 +187,12 @@ public BuildChain build() throws ChainBuildException {
                             final Throwable cause = new Throwable("This is the location of the conflicting producer ("
-                                    + toBeAdded.getStepBuilder().getBuildStep() + ")");
+                                    + toBeAdded.getStepBuilder().getBuildStep()
+                                    + ").  Use -Dquarkus.builder.log-conflict-cause=true to see the full stacktrace.");
```
Isn' t this message being displayed if `-Dquarkus.builder.log-conflict-cause=true` is set already?

Postremus replied:
I changed it so that this line is included conditionally based on LOG_CONFLICT_CAUSING.

Postremus committed:
Only print stacktrace of conflicting buildsteps if wanted
The quarkus-core-builder detects conflicts between buildsteps, and throws exception for that.
This exception also included the stacktrace of the original caller of the build method. However, this pointed to a big lambda chain.
Since this stacktrace was created for each buildstep, this got costly fast.

Now, simply only include by default an empty ST, and add an option to include the full stacktrace.
